app/controllers/group_forms_controller.rb

```diff
@@ -41,18 +41,16 @@ def update
 
     form = Form.find(params[:id])
 
-    receiving_group = Group.find(group_select_params[:group])
-    @group_select = Forms::GroupSelect.new(group: receiving_group, form: form)
+    @group_select = Forms::GroupSelect.new(group_select_params.merge(form: form))
 
-    # TODO add spec for checking: compare @group with @group_select.group to check if it's changed
-    if @group.external_id == receiving_group.external_id
-      flash[:message] = "Form is already in this group."
-      render :edit
-    else
-      success_message = "'#{form.name}' has been moved to '#{receiving_group.name}'"
-      form.move_to_group(receiving_group.external_id)
+    if @group_select.valid?
+      receiving_group = Group.find(@group_select.group)
 
+      @group_form.update!(group: receiving_group)
```

Comment:
I changed the from the `form.move_to_group` because it doesn't seem worth having a whole method to do it.

Feel free to change back though.